packages/compass-global-writes/src/components/states/shard-key-correct.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useMemo } from 'react';
+import {
+  Banner,
+  BannerVariant,
+  Body,
+  css,
+  Link,
+  spacing,
+  Code,
+  Subtitle,
+  Label,
+  SpinLoader,
+  Button,
+} from '@mongodb-js/compass-components';
+import { connect } from 'react-redux';
+import {
+  ShardingStatuses,
+  unmanageNamespace,
+  type RootState,
+  type ShardKey,
+  type ShardZoneData,
+} from '../../store/reducer';
+import toNS from 'mongodb-ns';
+import { ShardZonesTable } from '../shard-zones-table';
+import { useConnectionInfo } from '@mongodb-js/compass-connections/provider';
+
+const nbsp = '\u00a0';
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[400],
+});
+
+const codeBlockContainerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[100],
+  maxWidth: '700px',
+});
+
+const paragraphStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[100],
+});
+
+export type ShardKeyCorrectProps = {
+  namespace: string;
+  shardKey: ShardKey;
+  shardZones: ShardZoneData[];
+  isUnmanagingNamespace: boolean;
+  onUnmanageNamespace: () => void;
+};
+
+export function ShardKeyCorrect({
+  namespace,
+  shardKey,
+  shardZones,
+  isUnmanagingNamespace,
+  onUnmanageNamespace,
+}: ShardKeyCorrectProps) {
+  const customShardKeyField = useMemo(() => {
+    return shardKey.fields[1].name;
+  }, [shardKey]);
+
+  const { atlasMetadata } = useConnectionInfo();
+
+  const sampleCodes = useMemo(() => {
+    const { collection, database } = toNS(namespace);
+    return {
+      findingDocuments: `use ${database}\ndb.${collection}.find({"location": "US-NY", "${customShardKeyField}": "<id_value>"})`,
+      insertingDocuments: `use ${database}\ndb.${collection}.insertOne({"location": "US-NY", "${customShardKeyField}": "<id_value>",...<other fields>})`,
```

Comment:
You can have symbols in the collection name that make dot notation property access break, we usually use the `["<escaped coll name>"]` instead because it's more robust